carla-carma-integration/carla_carma_bridge/src/carla_carma_bridge/carma_carla_route

```diff
@@ -121,25 +116,20 @@ def initialize():
     #Ensure that localization is complete before checking route information
     localized_sub = rospy.Subscriber('/localization/current_pose', PoseStamped, pose_cb)
     while rospy.is_shutdown() != True:  
-        if local_pos.pose.position.x != 0 and local_pos.pose.position.y != 0:
+        if received_pose == True:
             if available_routes == None:
                 print("No routes available.")
             else:
                 """Check if the selected route is one of the available routes. If it isn't,then print error statement"""
-                if check_selected_route(available_routes) == True:
+                if check_selected_route(available_routes):
                     route = selected_route
                     set_route(route.route_id)
                     break
 
                 else:
-                    print("Error: Invalid selected route")
-                    rospy.spin()
-                
-            
+                    print("Error: Invalid selected route")                
```

Comment:
Are we going to keep loop running even the selected route is not one of the available routes?